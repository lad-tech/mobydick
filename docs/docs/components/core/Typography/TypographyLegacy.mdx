import Legacy from "./assets/Legacy.png";

<img src={Legacy} height={600} />

Reusable text

## Example

```tsx
import {TypographyLegacy} from "@lad-tech/mobydick-core";

export const Example = () => {
  return (
    <TypographyLegacy font={'Regular-Primary-S'}>Example</TypographyLegacy>
  );
};

```
## `Props`

### [`TextInput Props`](https://reactnative.dev/docs/textinput#props)

Inherits [TextInput Props](https://reactnative.dev/docs/textinput#props).

### `font`

| Type                                                                                                             | DEFAULT             |
|:-----------------------------------------------------------------------------------------------------------------|:--------------------|
| [TypographyLegacyProp](https://github.com/lad-tech/mobydick/blob/main/packages/core/src/typography/types.ts#L49) | 'Regular-Primary-S' |

Font for text
