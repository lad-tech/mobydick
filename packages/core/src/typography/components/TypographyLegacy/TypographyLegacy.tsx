import {FC} from 'react';

import Text from '../../../basic/components/Text/Text';
import {useFont} from '../../hooks';
import {IStyledTextLegacyProps} from '../../types';

export const TypographyLegacy: FC<IStyledTextLegacyProps> = ({
  children,
  font = 'Regular-Primary-S',
  style,
  ...props
}) => {
  const {fontStyle} = useFont(font);

  return (
    <Text style={[fontStyle, style]} {...props}>
      {children}
    </Text>
  );
};
