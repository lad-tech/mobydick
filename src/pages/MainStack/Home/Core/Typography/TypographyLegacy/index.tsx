import {FlatListProps, ListRenderItem} from 'react-native';

import {
  defaultTextLightColor,
  FlatList,
  TFontColor,
  TFontSize,
  TFontWeight,
  TypographyLegacy,
  TypographyLegacyProp,
  useStyles,
  View,
} from '@shared/ui';
import getScreenStyles from '@shared/styles/getScreenStyles';

const getAllTypographyLegacy = (): TypographyLegacyProp[] => {
  const weights: TFontWeight[] = Object.keys(
    TFontWeight,
  ) as unknown as TFontWeight[];

  const sizes: TFontSize[] = Object.keys(TFontSize) as unknown as TFontSize[];
  const colors: TFontColor[] = Object.keys(defaultTextLightColor).map(name =>
    name.slice(4, name.length),
  ) as unknown as TFontColor[];

  const result: TypographyLegacyProp[] = [];
  weights.forEach(weight => {
    colors.forEach(color => {
      sizes.forEach(size => {
        result.push(`${weight}-${color}-${size}`);
      });
    });
  });

  return result;
};

const renderItem: ListRenderItem<TypographyLegacyProp> = ({item}) => (
  <TypographyLegacy font={item}>{item}</TypographyLegacy>
);

const keyExtractor: FlatListProps<TypographyLegacyProp>['keyExtractor'] =
  item => item;
const TypographyLegacyScreen = () => {
  const [styles] = useStyles(getScreenStyles);
  const data = getAllTypographyLegacy();

  return (
    <View style={styles.container}>
      <FlatList
        data={data}
        renderItem={renderItem}
        keyExtractor={keyExtractor}
        removeClippedSubviews={true}
      />
    </View>
  );
};

export default TypographyLegacyScreen;
