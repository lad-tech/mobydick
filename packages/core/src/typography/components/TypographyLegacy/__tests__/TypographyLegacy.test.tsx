import {render} from '@testing-library/react-native';

import {TypographyLegacy} from '../TypographyLegacy';

describe('@lad-tech/mobydick-core/TypographyLegacy', () => {
  it('renders default correctly', () => {
    const {toJSON} = render(<TypographyLegacy>My Text</TypographyLegacy>);
    expect(toJSON()).toMatchSnapshot();
  });
});
