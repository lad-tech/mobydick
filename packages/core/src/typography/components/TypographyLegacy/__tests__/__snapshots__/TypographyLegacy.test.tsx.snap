// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@lad-tech/mobydick-core/TypographyLegacy renders default correctly 1`] = `
<Text
  allowFontScaling={true}
  style={
    [
      {
        "color": "#20242D",
        "fontFamily": "Inter-Regular",
        "fontSize": 15,
        "lineHeight": 19,
        "minHeight": 19,
      },
      undefined,
    ]
  }
>
  My Text
</Text>
`;
